src/python/grpcio/grpc/_simple_stubs.py

```diff
@@ -221,9 +238,10 @@ def unary_unary(
         immediately if the connection is not ready at the time the RPC is
         invoked, or if it should wait until the connection to the server
         becomes ready. When using this option, the user will likely also want
-        to set a timeout. Defaults to False.
+        to set a timeout. Defaults to True.
       timeout: An optional duration of time in seconds to allow for the RPC,
-        after which an exception will be raised.
+        after which an exception will be raised. If timeout is unspecified and
+        wait_for_ready is True, defaults to one minute.
```

Comment:
How to set `wait_for_ready=True` and `timeout` to infinite? Should we worry about this case? One use case might be a IoT scenario that the edge could automatically reconnect to master.

Reply:
I thought about this. Ideally, I'd like no value being specified to result in the default timeout introduced here and some sentinel value (perhaps `None`) to result in an infinite wait. The problem is that explicit keyword arguments don't let you differentiate between an unspecified argument and an argument specified with the value `None`. 

How about
```python
HelloService.SayHello(HelloRequest("you")) # wait_for_ready=True, timeout=DEFAULT
HelloService.SayHello(HelloRequest("you"), timeout=grpc.NoTimeout) # wait_for_ready=True, timeout=inf
```

Thoughts?

Reply:
Both solution looks fine to me. I slightly prefer the former one, because it doesn't expand our exposed API. Depends on you.

Reply:
By "former", you mean distinguishing between no argument specified (default timeout) and an explicit `None` (infinite timeout)?

Reply:
Yes. This one:
```
HelloService.SayHello(HelloRequest("you")) # wait_for_ready=True, timeout=DEFAULT
```

Reply:
While I like the interface here, the only way to implement this hurts the overall readibility of the code (both the gRPC surface and generated code).

```python
def unary_unary(...,
     wait_for_ready=Optional[bool],
      **kwargs):
  if "timeout" in kwargs:
    do_one_thing()
  else:
    do_another()
```
I do agree with you that it's better to keep the semantics of the existing API where possible, but the required change in signature seems like a net negative.

Reply:
If we want to provide type annotation for our users, we will need to expose a `grpc.DEFAULT` (or something like this). Then, it is equivalent or slightly worse than exposing an extra `grpc.NoTimeout`. `grpc.NoTimeout` semantically suggests it is a timeout value, while `grpc.DEFAULT` is vague. Sure, we can discuss this later today.